Fixes #1072 Current selected reminder not saved when editing a reminder

turtles2 opened the pull request:
Fixes #1072 Current selected reminder not saved when editing a reminder

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/monicahq/monica?pullRequest=1090) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/monicahq/monica?pullRequest=1090) before we can accept your contribution.<br/>**3** out of **8** committers have signed the CLA.<br/><br/>:white_check_mark: turtles2<br/>:white_check_mark: djaiss<br/>:white_check_mark: asbiin<br/>:x: hexnet<br/>:x: mwalling<br/>:x: JakariaBlaine<br/>:x: Kovah<br/>:x: henk23<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/monicahq/monica?pullRequest=1090) it.</sub>

turtles2 commented:
@djaiss or @asbiin Can we get a review on this PR?

asbiin commented on resources/views/people/reminders/form.blade.php:
```diff
@@ -50,15 +57,33 @@
+                        @if($update_or_add == 'edit' && $reminder->frequency_number != 'one_time')
+
+                            @if($reminder->frequency_type == 'week')
+                                <option selected value="week">{{ trans('people.reminders_type_week') }}</option>
```
You can do once:
```htm
<option value="week" {{ $update_or_add == 'edit' && $reminder->frequency_type == 'week' ? 'selected : '' }}>{{ trans('people.reminders_type_week') }}</option>
<option value="month" {{ $update_or_add == 'edit' && $reminder->frequency_type == 'month' ? 'selected : '' }}>{{ trans('people.reminders_type_month') }}</option>
<option value="year" {{ $update_or_add == 'edit' && $reminder->frequency_type == 'year' ? 'selected : '' }}>{{ trans('people.reminders_type_year') }}</option>
```

djaiss commented on resources/views/people/reminders/form.blade.php:
```diff
@@ -50,15 +57,15 @@
-                        <option value="week">{{ trans('people.reminders_type_week') }}</option>
-                        <option value="month">{{ trans('people.reminders_type_month') }}</option>
-                        <option value="year">{{ trans('people.reminders_type_year') }}</option>
+                        <option value="week" {{ $update_or_add == 'edit' && $reminder->frequency_type == 'week' ? 'selected : '' }}>{{ trans('people.reminders_type_week') }}</option>
```
This leads to an error. A `'` is missing after `selected`.
```
{{ $update_or_add == 'edit' && $reminder->frequency_type == 'week' ? 'selected HERE : '' }}
```

The same error appears on the 2 lines below.

turtles2 commented:
Sorry about that I have been making quick edits over GitHub so I have not
been able to run the changes.

On Tue, Apr 17, 2018, 10:26 PM Régis Freyd <notifications@github.com> wrote:

> *@djaiss* requested changes on this pull request.
> ------------------------------
>
> In resources/views/people/reminders/form.blade.php
> <https://github.com/monicahq/monica/pull/1090#discussion_r182299651>:
>
> >                             min="1"
>                             max="115"
>                             :disabled="reminders_frequency == 'once'">
>
>                      <select name="frequency_type" :disabled="reminders_frequency == 'once'">
> -                        <option value="week">{{ trans('people.reminders_type_week') }}</option>
> -                        <option value="month">{{ trans('people.reminders_type_month') }}</option>
> -                        <option value="year">{{ trans('people.reminders_type_year') }}</option>
> +                        <option value="week" {{ $update_or_add == 'edit' && $reminder->frequency_type == 'week' ? 'selected : '' }}>{{ trans('people.reminders_type_week') }}</option>
>
> This leads to an error. A ' is missing after selected.
>
> {{ $update_or_add == 'edit' && $reminder->frequency_type == 'week' ? 'selected HERE : '' }}
>
> The same error appears on the 2 lines below.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/monicahq/monica/pull/1090#pullrequestreview-113066877>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADfAujMUI1HzIkYJJ77cizFbCySP7a9Sks5tprJygaJpZM4TDAxj>
> .
>

turtles2 commented:
@djaiss Hope you don't mind but I added an account to the heroku instance so that I could test my changes. I have fixed it and it appears to be working.

djaiss commented:
@turtles2 thanks!

Unfortunately it doesn't work.
* Create a one-time reminder
* Try edit it. First of all it has become a recurrent reminder, and we can't change the value back.

This is due because of `:onload="reminders_frequency = 'recurrent'"` that you've added.

I don't think we should spend more time on this. I planned to rewrite this page anyway and make it much better.
